default-recommendations/for-loop-shortcuts-test.rkt

```diff
@@ -909,3 +909,73 @@ test: "(unless ...) in a for* loop refactored to #:when clause"
        #:unless (number? x))
   (displayln x))
 ------------------------------------------------------------
+
+
+test: "for/vector with in-range 0 n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for/vector ([i (in-range 0 n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for/vector #:length n
+            ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
+
+
+test: "for/vector with in-range n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for/vector ([i (in-range n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for/vector #:length n
+            ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
+
+
+test: "for*/vector with in-range 0 n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for*/vector ([i (in-range 0 n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for*/vector #:length n
+             ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
```

Comment:
Don't include this test case, and don't bother handling `for*/vector` in your new rule's implementation. All `for*` loops only do something different from their equivalent `for` loop when they're given more than one clause, making them redundant if there's only one clause. Just leave them alone for this pull request.